rocks-lib/src/config/mod.rs

```diff
@@ -33,6 +35,32 @@ impl FromStr for LuaVersion {
     }
 }
 
+impl TryFrom<&Rockspec> for LuaVersion {
+    type Error = eyre::Report;
+
+    fn try_from(rockspec: &Rockspec) -> std::result::Result<Self, Self::Error> {
+        let lua = rockspec
+            .dependencies
+            .current_platform()
+            .iter()
+            .find(|val| *val.name() == "lua".into())
+            .ok_or_eyre("no `lua` dependency found!")?;
+
+        for (possibility, version) in [
+            ("5.4.0", LuaVersion::Lua54),
+            ("5.3.0", LuaVersion::Lua53),
+            ("5.2.0", LuaVersion::Lua52),
+            ("5.1.0", LuaVersion::Lua51),
+        ] {
+            if lua.version_req().matches(&possibility.parse().unwrap()) {
+                return Ok(version);
+            }
+        }
+
+        Err(eyre!("no valid matches for `lua` found!"))
+    }
+}
+
```

Comment:
question: I'm completely torn on this. Should the `Rockspec` have a `.lua_version()` getter, or should we keep it like this? This approach allows the use of `rockspec.try_into()` as well as `let version = LuaVersion::from(&rockspec);`.

I feel like this is a bit more fitting as it performs a direct conversion into a `LuaVersion`, but I know of no best practices for these sorts of decisions.

Reply:
From the docs, [When to implement `From`](https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from):

> - The conversion is lossless: semantically, it should not lose or discard information.
> - The conversion is value-preserving: the conceptual kind and meaning of the resulting value is the same, even though the Rust type and technical representation might be different.
> - The conversion is obvious: it’s the only reasonable conversion between the two types. Otherwise it’s better to have it be a named method or constructor

So I'd argue that a getter is the better choice here. 

Also, I'd suggest a getter that returns an `Option` instead of a `Result`, as some rockspecs don't have a `dependencies` table.